File: BackEnd/src/main/java/com/spring/model/Employee.java
package com.spring.model;

import com.spring.model.Enums.Gender;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "employee")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Employee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "employee_id")
    private int id;
    @Column(name = "employee_name")
    private String name;
    @Column(name = "employee_age")
    private String age;
    @Column(name = "employee_address")
    private String address;
    @Column(name = "employee_phone")
    private String phone;
    @Column(name = "employee_createdDate")
    @CreationTimestamp
    private Date createdDate;
    @Column(name = "employee_lastUpdateDate")
    @UpdateTimestamp
    private Date lastUpdateDate;
    @Column(name = "employee_gender")
    private Gender gender;

}

File: FrontEnd/src/app/model/employee.ts
export class Employee {

  id : number;
  name : string;
  gender : string;
  age : string;
  phone : string;
  address : string;
  createdDate : Date;
  lastUpdateDate : Date;


  constructor(id: number, name: string, gender: string, age: string, phone: string, address: string, createdDate: Date, updatedDate: Date) {
    this.id = id;
    this.name = name;
    this.gender = gender;
    this.age = age;
    this.phone = phone;
    this.address = address;
    this.createdDate = createdDate;
    this.lastUpdateDate = updatedDate;
  }
}
